components/support/utils/src/main/java/mozilla/components/support/utils/DomainMatcher.kt

```diff
@@ -62,17 +62,19 @@ private fun matchSegment(query: String, rawUrl: String): String? {
         return rawUrl
     }
 
-    val url = URL(rawUrl)
-    if (url.host.startsWith(query)) {
-        return url.host + url.path + url.port.orEmpty()
-    }
+    val url = Uri.parse(rawUrl)
+    url.host?.let { host ->
```

Comment:
nit: I usually prefer something like:

```Kotlin
val host = url.host ?: return null
```

.. instead of a block causing indentation. :)